aptos-move/block-executor/src/view.rs

```diff
@@ -220,6 +231,12 @@ impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable> LatestView<
         }
         ret
     }
+
+    #[allow(dead_code)]
+    fn generate_id(&mut self) -> u64 {
+        self.id_counter.fetch_add(1, Ordering::SeqCst);
```

Comment:
fetch_add already returns the prior value, which is fine in this case.

If we do this properly (not using Sync data-structures in sequential path), we'd match the self.latest_view (see line 231 old fild), then increase the corresponding counter.